### DISCUSSION TITLE
Visualiser for analysed entities

### DISCUSSION THREAD

**Discussion by smileybear** [microsoft/presidio#1026] [General]
I wonder if there are plans for implementing some sort of visualiser (similar to displacy for spacey) for expanding understanding in what presidio is doing. 

Thanks

https://demos.explosion.ai/displacy-ent

**Comment by omri374** [comment]
Hi, we have a presidio online demo here: https://aka.ms/presidio-demo
In addition, there's a sample on how to use Streamlit to create a Presidio demo: https://microsoft.github.io/presidio/samples/python/streamlit/

Does this support your needs?

**Reply by smileybear** [reply]
Not in its entirety, say what if I am to create a custom recogniser, where would I add the deny and context lists within the py file in streamlit. Also visualisations across different colour codes would be appreciated as well. 
Can you expand the streamlit python file to contain further functionality ?

**Reply by omri374** [reply]
This is a cool visualization done by @benkilimnik: https://huggingface.co/spaces/beki/pii-anonymizer
The source code is available here: https://huggingface.co/spaces/beki/pii-anonymizer/blob/main/app.py
You could tweak the flow by changing how the AnalyzerEngine is loaded and called (with additional recognizers, parameters etc.)
HTH

**Reply by omri374** [reply]
Hi, our new demo website on HuggingFace hub has new functionality, including different color codes: https://huggingface.co/spaces/presidio/presidio_demo
